serde/example_test.go

```diff
@@ -130,3 +144,45 @@ func (exampleJSONFormat) Decode(ctx serde.Context, data []byte) (serde.Message,
 
 	return msg, nil
 }
+
+// exampleMessageXML is an XML message for a message example.
+type exampleMessageXML struct {
+	Value int `xml:"value"`
+}
+
+// exampleXMLFormat is an example if a format to serialize a message example
+// using the XML encoding.
+//
+// - implements serde.FormatEngine
+type exampleXMLFormat struct{}
+
+// Encode implements serde.FormatEngine. It formats the message to comply with
+// the XML encoding and marshal it.
+func (exampleXMLFormat) Encode(ctx serde.Context, msg serde.Message) ([]byte, error) {
+	example, ok := msg.(exampleMessage)
+	if !ok {
+		return nil, errors.New("unsupported message")
+	}
+
+	m := exampleMessageXML{
+		Value: example.value,
+	}
+
+	return ctx.Marshal(m)
+}
+
+// Decode implements serde.FormatEngine. It populates a message example if
+// appropritate, otherwise it returns an error.
+func (exampleXMLFormat) Decode(ctx serde.Context, data []byte) (serde.Message, error) {
+	var m exampleMessageXML
+	err := ctx.Unmarshal(data, &m)
+	if err != nil {
+		return nil, err
+	}
+
+	msg := exampleMessage{
+		value: m.Value,
+	}
+
+	return msg, nil
+}
```

Comment:
why do this and not use the same `exampleMessageEncoded` (merging `exampleMessageXML` and `exampleMessageJSON`) with an attribute  ``` `json:"value" xml:"value"` ``` ?

Reply:
While it would work, the goal is to show you can split entirely the logic of the formats because in actual use cases, there is a format implementation in an isolated package.

Reply:
Fair enough. Thanks for the clarification :)